controllers/etcd_controller.go

```diff
@@ -769,6 +775,144 @@ func (r *EtcdReconciler) getStatefulSetFromEtcd(etcd *druidv1alpha1.Etcd, values
 	return decoded, nil
 }
 
+func (r *EtcdReconciler) reconcileCronJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, values map[string]interface{}) (*batchv1.CronJob, error) {
+	logger.Info("Reconcile etcd compaction cronjob")
+
+	var cronJob batchv1.CronJob
+	err := r.Get(ctx, types.NamespacedName{Name: fmt.Sprintf("%s-compact-cronjob", etcd.Name), Namespace: etcd.Namespace}, &cronJob)
+
+	//If compactionSchedule is present in the etcd spec, continue with reconciliation of cronjob
+	//If compactionSchedule is not present in the etcd spec, do not proceed with cronjob
+	//reconciliation. Furthermore, delete any already existing cronjobs corresponding with this etcd
+	compactionScheduleFound := false
+	if values["backup"].(map[string]interface{})["compactionSchedule"] != nil {
+		compactionScheduleFound = true
+	}
+
+	if !compactionScheduleFound {
+		if err == nil {
+			err = r.Delete(ctx, &cronJob, client.PropagationPolicy(metav1.DeletePropagationBackground))
```

Comment:
Should propagation be foreground?

Reply:
Done
Yes, foreground makes better sense